File: app_java/src/main/java/vlab/android/architecture/di/AppComponent.java
package vlab.android.architecture.di;

import javax.inject.Singleton;

import dagger.Component;
import dagger.android.AndroidInjectionModule;
import vlab.android.architecture.MyApplication;

@Singleton
@Component(
        modules = {
                AndroidInjectionModule.class,
                AppModule.class
        }
)
public interface AppComponent {

//    @Component.Builder
//    interface Builder {
//        @BindsInstance Builder application(Application application);
//        @BindsInstance Builder appModule(AppModule appModule);
//        //@BindsInstance Builder networkModule(NetworkModule networkModule);
//        AppComponent build();
//    }

    void inject(MyApplication application);
}

File: app_kotlin/src/main/java/vlab/android/architecture/di/AppComponent.kt
package vlab.android.architecture.di

import dagger.Component
import dagger.android.AndroidInjectionModule
import vlab.android.architecture.MyApplication
import javax.inject.Singleton

@Singleton
@Component(modules = arrayOf(
        AndroidInjectionModule::class,
        AppModule::class)
)
interface AppComponent {
    fun inject(application: MyApplication)
}
